tortoise/models.py

```diff
@@ -594,6 +610,7 @@ class Model(metaclass=ModelMeta):
 
     # I don' like this here, but it makes auto completion and static analysis much happier
     _meta = MetaInfo(None)  # type: ignore
+    _listeners: Dict[Signals, Dict[Type[MODEL], List[Callable]]] = {}
```

Comment:
mypy complains about this. I thought about it a bit, and it's hint is just right.
Change the class definition to be:
```py
class Model(Generic[MODEL], metaclass=ModelMeta):
```
This will bound MODEL to the class of Model (not just the instance)